### DISCUSSION TITLE
Rework patterned field representation

### DISCUSSION THREAD

**Discussion by Blipz** [Azgaar/Armoria#174] [Ideas]
It would be nice to have more control on the sizing and position of patterns in a patterned/semy field.

Ideally `"t1": "barry-argent-gules-small"` could be replaced by something like `"field": {"pattern": "barry", "t1": "argent", "t2": "gules", "size": 0.8, "x": 0, "y": 0}`.

The COA string would become quite verbose, though, especially if `"t1": "azure"` gets changed to `"field": {"t1": "azure"}` for consistency reasons (and the same would apply for patterned charges and ordinaries). Also, it would break all existing links.

Backward-compatible alternatives could be to use something like `"t1": "barry-argent-gules-0.8|0|0"` or `"t1": "barry-argent-gules", "pattern": {"size": 0.8, "x": 0, "y": 0}`, but I find this quite ugly.

**Comment by Azgaar** [comment]
Makes sense
